picard/ui/options/scripting.py

```diff
@@ -103,18 +120,142 @@ class ScriptingOptionsPage(OptionsPage):
         self.ui.setupUi(self)
         self.highlighter = TaggerScriptSyntaxHighlighter(self.ui.tagger_script.document())
-        self.ui.tagger_script.textChanged.connect(self.live_checker)
+        self.ui.tagger_script.textChanged.connect(self.live_update_and_check)
+        self.ui.add_script.clicked.connect(self.add_to_lscript)
+        self.ui.remove_script.clicked.connect(self.remove_from_lscript)
+        self.ui.up_script.clicked.connect(self.move_script_up)
+        self.ui.down_script.clicked.connect(self.move_script_down)
+        self.ui.script_list.itemSelectionChanged.connect(self.script_selected)
+        self.ui.script_list.itemChanged.connect(self.script_attr_changed)
+        self.ui.tagger_script.setEnabled(False)
+        self.listitem_to_scriptitem = {}
+        self.list_of_scripts = []
+
+    def script_attr_changed(self, item):
+        item.setSelected(True)
+        script = self.listitem_to_scriptitem[item]
+        if item.checkState():
+            script.enabled = True
+        else:
+            script.enabled = False
+        script.name = item.text()
+        self.list_of_scripts[script.pos] = script.get_all()
+
+    def script_selected(self):
+        items = self.ui.script_list.selectedItems()
+        if items:
+            self.ui.tagger_script.setEnabled(True)
+            script = self.listitem_to_scriptitem[items[0]]
+            self.ui.tagger_script.setText(script.text_item)
+
+    def add_to_lscript(self):
+        count = self.ui.script_list.count()
+        script = ScriptItem(position=count, title=_(self.SCRIPT_DEFAULT_NAME)+" "+str(count+1))
+        list_item = QtGui.QListWidgetItem(script.name)
+        list_item.setFlags(list_item.flags() | QtCore.Qt.ItemIsUserCheckable | QtCore.Qt.ItemIsEditable)
+        list_item.setCheckState(QtCore.Qt.Checked)
+        self.ui.script_list.addItem(list_item)
+        self.listitem_to_scriptitem[list_item] = script
+        self.list_of_scripts.append(script.get_all())
+
+    def update_script_positions(self):
+        for i, script in enumerate(self.list_of_scripts):
+            self.list_of_scripts[i] = (i, script[1], script[2], script[3])
+            item = self.ui.script_list.item(i)
+            self.listitem_to_scriptitem[item].pos = i
+
+    def remove_from_lscript(self):
+        currentRow = self.ui.script_list.currentRow()
+        item = self.ui.script_list.takeItem(currentRow)
+        if item:
+            script = self.listitem_to_scriptitem[item]
+            del self.listitem_to_scriptitem[item]
+            del self.list_of_scripts[script.pos]
+            del script
+            item = None
+            # update positions of other items
+            self.update_script_positions()
+            if self.ui.script_list.count() == 0:
+                self.ui.tagger_script.setText("")
+                self.ui.tagger_script.setEnabled(False)
 
-    def live_checker(self):
+    def move_script_up(self):
```

Comment:
There is a lot of code duplication between `move_script_up()` and `move_script_down()`. Please improve.